resharper/resharper-unity/src/CSharp/Feature/Services/QuickFixes/MulOrderQuickFix.cs

```diff
@@ -0,0 +1,80 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Application.Progress;
+using JetBrains.ProjectModel;
+using JetBrains.ReSharper.Feature.Services.QuickFixes;
+using JetBrains.ReSharper.Plugins.Unity.CSharp.Daemon.Errors;
+using JetBrains.ReSharper.Psi;
+using JetBrains.ReSharper.Psi.CSharp;
+using JetBrains.ReSharper.Psi.CSharp.Tree;
+using JetBrains.ReSharper.Psi.CSharp.Util;
+using JetBrains.ReSharper.Psi.Tree;
+using JetBrains.TextControl;
+using JetBrains.Util;
+
+namespace JetBrains.ReSharper.Plugins.Unity.CSharp.Feature.Services.QuickFixes
+{
+    [QuickFix]
+    public class MulOrderQuickFix : QuickFixBase
+    {
+        private readonly IMultiplicativeExpression myExpression;
+
+        public MulOrderQuickFix(InefficientMultiplyOrderWarning warning)
+        {
+            myExpression = warning.Expression;
+        }
+        
+        protected override Action<ITextControl> ExecutePsiTransaction(ISolution solution, IProgressIndicator progress)
+        {
+            var operands = GetAllOperands(myExpression);
+
+            var numerics = operands.Where(t => t.GetExpressionType().ToIType().IsPredefinedNumeric()).ToList();
+            var vector = operands.Single(t => !t.GetExpressionType().ToIType().IsPredefinedNumeric());
```

Comment:
Is this guaranteed to be just one vector? Would it be easier to sort based on `IsPredefinedNumeric` with `true` coming first and then just loop over all operands?